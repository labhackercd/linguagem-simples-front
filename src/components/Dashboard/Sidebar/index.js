import React from 'react';
import { Grid } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import  UserIcon  from './../../../assets/User.svg';
import  HomeIcon  from './../../../assets/Home.svg';
import  ACIcon  from './../../../assets/AC.svg';
import  ExitIcon  from './../../../assets/Exit.svg';

const useStyles = makeStyles((theme) => ({
  sidebar: {
  	backgroundColor: "#00AF82",
  	display: 'flex',
  	flexDirection: 'column',
  	justifyContent: 'space-between',
  	height: '100%',
  },
  user: {
  	display: 'flex',
  	alignSelf: 'center',
  	margin: '1rem 1rem 0 0'
  },
  middleSection: {
  	display: 'flex',
  	flexDirection: 'column',
  	justifyContent: 'space-around',
  	alignItems: 'center',
  },
  middleIcon: {
  	padding: '1rem 0 0 0'
  },
  exit: {
  	display: 'flex',
  	alignSelf: 'center',
    alignContent: 'flex-end',
  },
  exitIcon: {
    display: 'flex',
    alignSelf: 'flex-end',
    margin: '0 0 1rem 0'
  }
}));

export default function Sidebar() {
  const classes = useStyles();
  return (
		<Grid container className={classes.sidebar}>
			<Grid item md={3} className={classes.user}>
				<a href="/"><img src={UserIcon} alt="user icon"/></a>
			</Grid>
			<Grid container className={classes.middleSection}>
				<Grid item md={6} className={classes.middleIcon}>
					<a href="/"><img src={HomeIcon} alt="home icon"/></a>
				</Grid>
				<Grid item md={6} className={classes.middleIcon}>
					<a href="/"><img src={ACIcon} alt="AC icon"/></a>
				</Grid>
			</Grid>
			<Grid item md={3} className={classes.exit}>
        <div className={classes.exitIcon}>
				    <a href="/"><img src={ExitIcon} alt="Exit icon"/></a>
        </div>
			</Grid>
		</Grid>
  )
}
