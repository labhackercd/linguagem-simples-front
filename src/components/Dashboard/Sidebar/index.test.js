import React from 'react';
import { render, screen } from '@testing-library/react';
import Sidebar from './index';
import {shallow} from "enzyme/build";

it("should render the content section", () => {
    const component = shallow(<Sidebar/>);
    expect(component).toMatchSnapshot();
});
