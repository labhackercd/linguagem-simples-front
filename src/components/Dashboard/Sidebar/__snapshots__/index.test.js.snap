// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the content section 1`] = `
<WithStyles(ForwardRef(Grid))
  className="makeStyles-sidebar-1"
  container={true}
>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-user-2"
    item={true}
    md={3}
  >
    <a
      href="/"
    >
      <img
        alt="user icon"
        src="../../img/User.svg"
      />
    </a>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-middleSection-3"
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-middleIcon-4"
      item={true}
      md={6}
    >
      <a
        href="/"
      >
        <img
          alt="home icon"
          src="../../img/Home.svg"
        />
      </a>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      className="makeStyles-middleIcon-4"
      item={true}
      md={6}
    >
      <a
        href="/"
      >
        <img
          alt="AC icon"
          src="../../img/AC.svg"
        />
      </a>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-exit-5"
    item={true}
    md={3}
  >
    <div
      className="makeStyles-exitIcon-6"
    >
      <a
        href="/"
      >
        <img
          alt="Exit icon"
          src="../../img/Exit.svg"
        />
      </a>
    </div>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
